model/credential.py

```diff
@@ -89,14 +85,40 @@ class Credential(Base):
     IDENTIFIER_FROM_REMOTE_SERVICE = "Identifier Received From Remote Service"
 
     @classmethod
-    def lookup(self, _db, data_source, type, patron, refresher_method,
+    def _filter_out_credential(cls, credential, allow_persistent_token):
```

Comment:
Putting this in a helper method is good; can you rename this to something like `_filter_invalid_credential`? Currently it sounds like the credential is _always_ filtered out.

Reply:
Updated the method's name